lib/vagrant.pm

```diff
@@ -0,0 +1,34 @@
+package vagrant;
+use testapi;
+use strict;
+use warnings;
+use utils;
+
+our @ISA    = qw(Exporter);
+our @EXPORT = qw(setup_vagrant);
+
+# - install vagrant and all additional packages
+# - launch the required daemons
+#
+# Named parameters:
+# - install_virtualbox: bool
+#   whether to install & setup virtualbox instead of libvirt
+sub setup_vagrant {
+    my ($args) = @_;
+
+    select_console('root-console');
+
+    my $extra_pkgs  = 'vagrant-libvirt';
+    my $daemon_name = 'libvirtd';
+    my $user_group  = 'vboxusers';
+
+    if ($args->{install_virtualbox}) {
+        $extra_pkgs  = 'virtualbox';
+        $daemon_name = 'vboxdrv vboxautostart';
+        $user_group  = 'vboxusers';
+    }
+
+    zypper_call("in vagrant $extra_pkgs");
+    assert_script_run("systemctl start $daemon_name");
+    assert_script_run("usermod -a -G $user_group bernhard");
```

Comment:
https://openqa.opensuse.org/tests/1007265#step/tumbleweed/8 fails because `vboxusers` is installed only when `virtualbox` is installed. And you do not need to add the user to this group for the `libvirt` case.

Reply:
Yep, copy+paste fail from my side.

Reply:
New test run: https://openqa.opensuse.org/tests/1007368

Reply:
> New test run: https://openqa.opensuse.org/tests/1007368

Seems better, but still some (unrelated?) issues.

Reply:
Next try with the most recent snapshot: https://openqa.opensuse.org/tests/1011218

Reply:
openQA is happy now

Reply:
> openQA is happy now

Not sure if we could replace `bernhard` with a variable?
Anyway, we can merge this PR.

Reply:
Hm, I thought that user was actually hardcoded somewhere, but I am not sure about that (I've seen it used over and over, so I assumed that is somehow standard)